internal/tools/aws/database_multitenant.go

```diff
@@ -0,0 +1,728 @@
+// needs to perform non-complex queries in a SQL database instance.
+type SQLDatabaseManager interface {
+	QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error)
+	Close() error
+}
+
+// RDSMultitenantDatabase is a database backed by RDS that supports multi-tenancy.
+type RDSMultitenantDatabase struct {
+	client         *Client
+	db             SQLDatabaseManager
+	installationID string
+}
+
+// NewRDSMultitenantDatabase returns a new instance of RDSMultitenantDatabase that implements database interface.
+func NewRDSMultitenantDatabase(installationID string, client *Client) *RDSMultitenantDatabase {
+	return &RDSMultitenantDatabase{
+		client:         client,
+		installationID: installationID,
+	}
+}
+
+// Teardown removes all AWS resources related to a RDS multitenant database.
+func (d *RDSMultitenantDatabase) Teardown(store model.InstallationDatabaseStoreInterface, keepData bool, logger log.FieldLogger) error {
+	databaseName := MattermostRDSDatabaseName(d.installationID)
+
+	logger = logger.WithField("rds-multitenant-database", databaseName)
+	logger.Info("Tearing down RDS database and database secret")
+
+	multitenantDatabase, err := store.GetMultitenantDatabaseForInstallationID(d.installationID)
+	if err != nil {
+		return errors.Wrapf(err, "unable to find multitenant database for installation ID %s", d.installationID)
+	}
+
+	err = d.removeRDSDatabase(multitenantDatabase, databaseName, store, logger)
+	if err != nil {
+		return errors.Wrapf(err, "unable to finish teardown of multitenant database name %s", databaseName)
+	}
+
+	multitenantDatabaseSecretName := RDSMultitenantSecretName(d.installationID)
+
+	_, err = d.client.Service().secretsManager.DeleteSecret(&secretsmanager.DeleteSecretInput{
+		SecretId: aws.String(multitenantDatabaseSecretName),
+	})
+	if err != nil && !IsErrorCode(err, secretsmanager.ErrCodeResourceNotFoundException) {
+		return errors.Wrapf(err, "failed to delete multitenant database secret name %s", multitenantDatabaseSecretName)
+	}
+
+	logger.Infof("Multitenant RDS database teardown complete")
+
+	return nil
+}
+
+// Snapshot creates a snapshot of single RDS multitenant database.
+func (d *RDSMultitenantDatabase) Snapshot(store model.InstallationDatabaseStoreInterface, logger log.FieldLogger) error {
+	return errors.New("not implemented")
+}
+
+// GenerateDatabaseSpecAndSecret creates the k8s database spec and secret for accessing a single database inside
+// a RDS multitenant cluster.
+func (d *RDSMultitenantDatabase) GenerateDatabaseSpecAndSecret(store model.InstallationDatabaseStoreInterface, logger log.FieldLogger) (*mmv1alpha1.Database, *corev1.Secret, error) {
+	installationDatabaseName := MattermostRDSDatabaseName(d.installationID)
+
+	logger = logger.WithField("rds-multitenant-database", installationDatabaseName)
+
+	multitenantDatabases, err := store.GetMultitenantDatabases(&model.MultitenantDatabaseFilter{
```

Comment:
Feel free to use the new method you created on this. `GetMultitenantDatabaseForInstallationID`